server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -235,7 +235,9 @@ public final class IndexScopedSettings extends AbstractScopedSettings {
             IndexMetadata.INDEX_REMOTE_TRANSLOG_STORE_ENABLED_SETTING,
             IndexMetadata.INDEX_REMOTE_TRANSLOG_REPOSITORY_SETTING,
             IndexMetadata.INDEX_REMOTE_TRANSLOG_BUFFER_INTERVAL_SETTING
-        )
+        ),
+        FeatureFlags.CONCURRENT_SEGMENT_SEARCH,
+        List.of(IndexSettings.INDEX_CONCURRENT_SEGMENT_SEARCH_SETTING)
```

Comment:
Can we also add some test to validate when FeatureFlag is not enabled, updating these settings fails ? Similar to ones in `IndexSettingsTests`

Reply:
Looking at `FEATURE_FLAGGED_INDEX_SETTINGS` it looks like we actually don't have any tests for any of the existing settings.

For the test `IndexSettingsTests.testDefaultSearchPipelineWithoutFeatureFlag` it's actually testing the settings update consumer (which we just removed in the latest revision).

I did do some manual testing just to confirm:
```
dengjay@88665a3f24d4 opensearch-project-dev % curl -X PUT "localhost:9200/my-index-000001/_settings?pretty" -H 'Content-Type: application/json' -d'
{
  "index" : {
    "search.concurrent_segment_search.enabled" : null
  }
}
'
{
  "error" : {
    "root_cause" : [
      {
        "type" : "settings_exception",
        "reason" : "unknown setting [index.search.concurrent_segment_search.enabled] please check that any required plugins are installed, or check the breaking changes documentation for removed settings"
      }
    ],
    "type" : "settings_exception",
    "reason" : "unknown setting [index.search.concurrent_segment_search.enabled] please check that any required plugins are installed, or check the breaking changes documentation for removed settings"
  },
  "status" : 400
}
```
```
dengjay@88665a3f24d4 opensearch-project-dev % curl -X PUT "localhost:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'        
{
  "persistent" : {
    "search.concurrent_segment_search.enabled" : null   
  }
}
'
{
  "error" : {
    "root_cause" : [
      {
        "type" : "settings_exception",
        "reason" : "persistent setting [search.concurrent_segment_search.enabled], not recognized"
      }
    ],
    "type" : "settings_exception",
    "reason" : "persistent setting [search.concurrent_segment_search.enabled], not recognized"
  },
  "status" : 400
}
```

I think I can add this to `SettingsModuleTests` instead though, taking a look.

Reply:
Added these test cases to `SettingsModuleTests` in the latest revision